docs/user/exercises/programming-exercise-AI-proof-guidelines.inc

```diff
@@ -0,0 +1,491 @@
+              this.type = type;
+              this.weight = weight;
+              face = "";
+              seeds = true;
+          }
+
+      }
+
+
+-  **Depictions of functionalities:** Rather than explaining how a
+   concept works in theory, provide depictions of the functionality.
+   This encourages students to understand the basic code instead of
+   going to AI for answers. Examples that were used in previous
+   exercises:
+
+
+   .. figure:: programming/AI-proof-exercises/List-generics.png
+      :align: center
+      :alt: This figure showcases the functionality of a linked-list
+      :width: 50%
+
+
+      Depiction shows the functionality of a linked list with 3 integer values
+      as a basic concept for understanding linked lists. The task of this exercise
+      were to implement the common List functionalities from scratch rather than
+      rely on the standard methods from java. ListNode and LinkedList were
+      therefore own classes that needed to be filled with the required
+      functionalities.
+
+
+   This depiction provides an understanding of how a linked list should look
+   like after the add function was called.
+
+   .. figure:: programming/AI-proof-exercises/add.png
+      :align: center
+      :width: 50%
+
+
+      Figure shows what a list should look like after the add functionality was called. Picture from the Problem statement
+
+-  **Text in picture format:** Giving the text exercise instructions in
+   a picture format that can not be copied. Rather than writing long
+   inquiries into the AI, the students will be promoted to asking
+   reasonable questions and solve exercises on their own.
+
+   .. note::
+      Since there are already possibilities to copy text from pictures, the effect of
+      this measure are limited. More suitable for exam mode.
+
+-  **Java frameworks:** Using Java frameworks, such as JavaFX, is challenging for AI
+   and helps to visualize projects in a more interactive way
+   for students through gamification. How can we incorporate that for
+   our exercises?
+
+   -  Provide Graphical User Interfaces (GUI) for
+      students with which they can also test the functionality of their
+      code.
+
+   -  Give exercises that use GUIs because AI has difficulty
+      solving them accurately.
+
+   -  Provide pictures of how the GUI should look in the end.
+
+      Students needed to implement the GUI based on the this figure that
+      was given in the problem statement.
+
+      .. figure:: programming/AI-proof-exercises/GUI-upgrade-for-game.png
+         :align: center
+         :width: 30%
+
+
+         Example GUI from homework exercise
+
+   .. note::
+      Apply this approach when students already have knowledge of what a GUI is. Most likely in
+      a later stage of the semester.
+
+.. note::
+   ChatGPT is limited to text input. Including a fair amount of graphical
+   representation for the exercises helps to limit the effectiveness of such tools.
```

Comment:
Only the free version. The paid version already has multi-modal capabilites to recognize what is in images. How good this is in our case I don't know

Reply:
Exactly. the paid version. I assume most students do not pay for GPT-4. It can still be evaded but it is harder and disencourages its use.

Reply:
Well. Google Bard is free and also has multi-modal capabilities to recognize images. So do several Llama 2 based models that are also freely accessible in limited ways on the internet.

I just mentioned GPT4, since it is the best known multi-modal model, but this point is still already outdated today.